staging/cert-manager-setup/templates/post-install-hook-job.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "cert-manager-setup.fullname" . }}
+  labels:
+{{ include "cert-manager-setup.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "-6"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "wait-for-cert-manager-webhook"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: {{ .Chart.Name }}
+        image: bitnami/kubectl:latest
+        imagePullPolicy: IfNotPresent
+        command: ["kubectl", "wait", "--for=condition=Available", "--timeout=300s", "apiservice", "v1beta1.webhook.certmanager.k8s.io"]
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "cert-manager-setup.fullname" . }}-sleep
+  labels:
+{{ include "cert-manager-setup.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "sleep"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: {{ .Chart.Name }}
+        image: ubuntu:xenial
+        imagePullPolicy: IfNotPresent
+        command: ["sleep", "30"]
```

Comment:
this is ugly :/ ... we need to check which resource we need to query to be sure the apiserver is ready. This could cause race conditions based on certain scenarios.

Reply:
if we need a sleep, it'll be better to create an `extra-step` to check the service directly. since you're within the cluster, it should be simple.

Reply:
Let's stop to confuse him ... using an extra-step again.

Reply:
what does your comment have to do with my comment about the `["sleep", "30"]` ?

Reply:
i suggested:

> you can replace `["kubectl", "wait", "--for=condition=Available", "--timeout=300s", "apiservice", "v1beta1.webhook.certmanager.k8s.io"]` to run the extra step binary instead.

i think you're sleeping for `30` seconds because you're not sure if the webhook is installed properly?

Reply:
It appears as if the original wait doesn't wait long enough. If I/you/we find a command that waits for the right thing the sleep could be removed again.

Reply:
please, file a jira ticket for that

Reply:
https://jira.mesosphere.com/browse/DCOS-58871